wasmbinding/queries.go

```diff
@@ -103,3 +107,40 @@ func (qp QueryPlugin) EstimateSwap(ctx sdk.Context, estimateSwap *bindings.Estim
 	estimate, err := PerformSwap(qp.gammKeeper, ctx, senderAddr, estimateSwap.ToSwapMsg())
 	return estimate, err
 }
+
+func (qp QueryPlugin) ArithmeticTwap(ctx sdk.Context, arithmeticTwap *bindings.ArithmeticTwap) (*sdk.Dec, error) {
+	if arithmeticTwap == nil {
+		return nil, wasmvmtypes.InvalidRequest{Err: "gamm arithmetic twap null"}
+	}
+
+	poolId := arithmeticTwap.PoolId
+	quoteAssetDenom := arithmeticTwap.QuoteAssetDenom
+	baseAssetDenom := arithmeticTwap.BaseAssetDenom
+	startTime := time.UnixMilli(arithmeticTwap.StartTime)
+	endTime := time.UnixMilli(arithmeticTwap.EndTime)
+
+	twap, err := qp.twapKeeper.GetArithmeticTwap(ctx, poolId, quoteAssetDenom, baseAssetDenom, startTime, endTime)
+	if err != nil {
+		return nil, sdkerrors.Wrap(err, "gamm arithmetic twap")
+	}
+
+	return &twap, nil
+}
+
+func (qp QueryPlugin) ArithmeticTwapToNow(ctx sdk.Context, arithmeticTwap *bindings.ArithmeticTwapToNow) (*sdk.Dec, error) {
+	if arithmeticTwap == nil {
+		return nil, wasmvmtypes.InvalidRequest{Err: "gamm arithmetic twap null"}
+	}
+
+	poolId := arithmeticTwap.PoolId
+	quoteAssetDenom := arithmeticTwap.QuoteAssetDenom
+	baseAssetDenom := arithmeticTwap.BaseAssetDenom
+	startTime := time.Unix(arithmeticTwap.StartTime, 0)
```

Comment:
ah we forgot to make this milli in the version we sent to mars

Reply:
I can spin one up fairly quickly if this needs to be changed